### ISSUE TITLE
Product Page v2 : Pre-defined feature values are populated with custom values

### ISSUE THREAD

**Issue by jf-viguier** [PrestaShop/PrestaShop#37543]
### Prerequisites

- [x] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [x] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

In product Page v2, the Pre-defined feature values are populated with custom values associated with other products

### Expected behavior

Pre-defined feature values qhould not be populated with custom values at all. It's a huge performance issue for large catalogs.

### Steps to reproduce

1. Fill-in some custom feature values for a feature X for some products
2. Go to a product Y and fill-in this feature with a custom value.
3. Reload this product page
4. you'll see that pre-defined values are populated with custom values, all custom values associated with at least a product

![Image](https://github.com/user-attachments/assets/c3ce1466-c998-4537-9d99-0fa362cb3718)

### PrestaShop version(s) where the bug happened

8.2

### PHP version(s) where the bug happened

all

### Your company or customer's name goes here (if applicable).

Creabilis.com

**Comment by Hlavtox** [comment]
@jf-viguier Can you check the redesigned feature inputs in v9? Maybe it's fixed there. 👍

**Comment by jf-viguier** [comment]
@Hlavtox yes it's fixed because of an ui change,  but the v9 is not ready for production.
Can you point me where the predefined list is populated ? I can't figure it out ?

**Comment by florine2623** [comment]
Hello @jf-viguier ,

I can't reproduce your issue :

https://github.com/user-attachments/assets/a17652da-24ca-47fc-8072-4bceae4f44cd

Am I missing something ?
Thanks!

**Comment by jf-viguier** [comment]
@florine2623 please follow the how to test, step 2, 3 and 4 : 
- Fill-in some custom feature values for a feature X for some products
- Go to a product Y and fill-in this custom feature.
- Reload this product page
- yo'll see that pre-defined values are populated with custom values

**Comment by Touxten** [comment]
Hello @florine2623 i can reproduce the issue.

Step Add ten customs value on a product, add 1 custom value on an other  product save the product and check the values on the select  

The select here must have predefined values and not custom values. This is a major bug that causes the product page to fall into small catalogs.

Best regards

**Comment by Touxten** [comment]
![Image](https://github.com/user-attachments/assets/679c8b45-6507-4e5d-b4cd-5a2ffaa40068)

**Comment by Touxten** [comment]
In other words, the number of requests multiplies by the number of features on a product, which quickly becomes enormous...

**Comment by jf-viguier** [comment]
Thanks @Touxten

**Comment by florine2623** [comment]
I still can't reproduce the issue, neither on 8.2.0 nor 9.0.0.

@paulnoelcholot could you try to reproduce the issue on PS 8.2.0 ?

Thanks!

**Comment by florine2623** [comment]
Actually, I have reproduced the issue.

![Image](https://github.com/user-attachments/assets/80b091f2-0091-4be1-9cc2-c7f25f821910)
![Image](https://github.com/user-attachments/assets/15b47df8-8f0c-4150-97bb-be9f6a844b86)

Tested on 2 products.
The pre-defined value should be empty.

I have not understood well the issue. I thought the pre-defined value of product 1 was displayed on product 2.

**Comment by boo-code** [comment]
Does not reproduce on 9.2. The endpoint that fills the predefined list excludes custom values in SQL.

`FeatureController::getAllFeatureGroupsAction()` calls `FeatureRepository::getFeaturesWithValues()`, whose query carries:

```php
->where('fv.custom = 0')
```

Measured on a 9.2 shop by adding two values to a shop associated feature, one predefined and one custom, then calling that method:

```
feature=1  predefined_value=19  custom_value=20

{"feature_id":1,"name":"Composition","values":[
  {"item_id":1,"name":"Polyester"}, {"item_id":2,"name":"Wool"}, ...,
  {"item_id":19,"name":"PREDEFINED-37543"}
]}

predefined value offered to the product page: YES
custom value offered to the product page:     no
```

The custom value never reaches the list, so neither the wrong values nor the catalogue size problem that came with them apply here. That matches what @jf-viguier reported after checking v9 and why @florine2623 could not reproduce it.

#37577 targeted `8.2.x` and is closed, so nothing is pending for the maintained branches.
